dapp/src/controllers/walletDetailCtrl.js

```diff
@@ -243,12 +253,17 @@
                 };
               case "a9059cbb":
                 var tokenAddress = tx.to;
-                var account = "0x" + tx.data.slice(34, 74);
+                var account = Web3Service.toChecksumAddress("0x" + tx.data.slice(34, 74));
                 var token = {};
                 Object.assign(token, $scope.wallet.tokens[tokenAddress]);
                 token.balance = new Web3().toBigNumber("0x" + tx.data.slice(74));
+                var filteredAddress = $filter("addressCanBeOwner")(account, $scope.wallet);
+                if (filteredAddress==account && $scope.addressBook && $scope.addressBook[filteredAddress]) {
```

Comment:
I would create another filter to check inside the address book for better readability